alerter.c

```diff
@@ -11,16 +16,44 @@ int networkAlertStub(float celcius) {
     return 200;
 }
 
-void alertInCelcius(float farenheit) {
+void testAlertFailure(int Failed_countInital,float TemperatureInCelcious,int networkReturnCode)
+{
+    if(TemperatureInCelcious>TEST_THRESHOLD_TEMPERATURE)
+    {
+        assert(networkReturnCode==500);
+        assert(Failed_countInital!=alertFailureCount);
+    }
+    else
+    {
+        assert(networkReturnCode==200);
+        assert(Failed_countInital==alertFailureCount);
+    }
+}
+
+#endif
+
+void alertInCelcius(float farenheit) 
+{
     float celcius = (farenheit - 32) * 5 / 9;
+    
+#if (TEST_CODE_ENABLED == STD_ON)

```

Comment:
where is the #elsif ? either test stub or production code has to execute. in this case only test stub is executed.

Test code should be abstracted neatly via an interface or a function pointer.
There is a danger of accidental execution of test code in field!!

Reply:
There is only stub available for the actual network function so that i added #if alone .
I will correct them